src/plans/higher_order_primal_dual_plan.jl

```diff
@@ -0,0 +1,234 @@
+    prox_G_dual,
+    diff_prox_G_dual,
+    linearized_forward_operator,
+    adjoint_linearized_operator;
+    Λ::Union{Function,Missing}=missing,
+    evaluation::AbstractEvaluationType=AllocatingEvaluation(),
+) where {mT<:AbstractManifold,nT<:AbstractManifold}
+    return PrimalDualSemismoothNewtonProblem{typeof(evaluation),mT,nT}(
+        M,
+        N,
+        cost,
+        prox_F,
+        diff_prox_F,
+        prox_G_dual,
+        diff_prox_G_dual,
+        linearized_forward_operator,
+        adjoint_linearized_operator,
+        Λ,
+    )
+end
+
+@doc raw"""
+    PrimalDualSemismoothNewtonOptions <: PrimalDualOptions
+
+* `m` - base point on $ \mathcal M $
+* `n` - base point on $ \mathcal N $
+* `x` - an initial point on $x^{(0)} \in \mathcal M$ (and its previous iterate)
+* `ξ` - an initial tangent vector $\xi^{(0)}\in T_{n}^*\mathcal N$ (and its previous iterate)
+* `primal_stepsize` – (`1/sqrt(8)`) proximal parameter of the primal prox
+* `dual_stepsize` – (`1/sqrt(8)`) proximal parameter of the dual prox
+* `reg_param` – (`1e-5`) regularisation parameter for the Newton matrix
+* `stop` - a [`StoppingCriterion`](@ref)
+* `update_primal_base` (`(p,o,i) -> o.m`) function to update the primal base
+* `update_dual_base` (`(p,o,i) -> o.n`) function to update the dual base
+* `retraction_method` – (`ExponentialRetraction()`) the rectraction to use
+* `inverse_retraction_method` - (`LogarithmicInverseRetraction()`) an inverse retraction to use.
+* `vector_transport_method` - (`ParallelTransport()`) a vector transport to use
+
+where for the last two the functions a [`Problem`](@ref) `p`,
+[`Options`](@ref) `o` and the current iterate `i` are the arguments.
+If you activate these to be different from the default identity, you have to provide
+`p.Λ` for the algorithm to work (which might be `missing`).
+
+# Constructor
+    PrimalDualSemismoothNewtonOptions(m::P, n::Q, x::P, ξ::T, primal_stepsize::Float64, dual_stepsize::Float64, reg_param::Float64;
+        stopping_criterion::StoppingCriterion = StopAfterIteration(50),
+        update_primal_base::Union{Function,Missing} = missing,
+        update_dual_base::Union{Function,Missing} = missing,
+        retraction_method = ExponentialRetraction(),
+        inverse_retraction_method = LogarithmicInverseRetraction(),
+        vector_transport_method = ParallelTransport(),
+    )
+"""
+mutable struct PrimalDualSemismoothNewtonOptions{
+    P,
+    Q,
+    T,
+    RM<:AbstractRetractionMethod,
+    IRM<:AbstractInverseRetractionMethod,
+    VTM<:AbstractVectorTransportMethod,
+} <: PrimalDualOptions
+    m::P
+    n::Q
+    x::P
+    ξ::T
+    primal_stepsize::Float64
+    dual_stepsize::Float64
+    reg_param::Float64
+    stop::StoppingCriterion
+    update_primal_base::Union{Function,Missing}
+    update_dual_base::Union{Function,Missing}
+    retraction_method::RM
+    inverse_retraction_method::IRM
+    vector_transport_method::VTM
+
+    function PrimalDualSemismoothNewtonOptions(
```

Comment:
Recently I switched the constructors for options to start with the manifold (similar to the problem). Though the manifold is not stored within the options, it enables to use the already mentioned `default_retraction_method(M)` and such for the three last warts you have here (and the docs should be adapted here as well for sure).

Reply:
changed

Reply:
This I do not see in the PR, did you commit/push it ;) ?

Reply:
I was too quick claiming this was fixed. Changed now :)

Reply:
Mo problem, that‘s exactly why one can comment here and mark things as resolved when both have finished the thing discussed.